crates/wascc-provider/src/lib.rs

```diff
@@ -122,20 +123,20 @@ impl StopHandler for ActorHandle {
 /// TODO: In the future, we will look at loading capabilities using the "sidecar" metaphor
 /// from Kubernetes.
 #[derive(Clone)]
-pub struct WasccProvider<S> {
+pub struct WasccProvider {
     handles: Arc<RwLock<HashMap<String, Handle<ActorHandle, LogHandleFactory>>>>,
-    store: S,
+    store: Arc<dyn Store + Sync + Send>,
     volume_path: PathBuf,
     log_path: PathBuf,
     kubeconfig: kube::Config,
     host: Arc<Mutex<WasccHost>>,
 }
 
-impl<S: Store + Send + Sync> WasccProvider<S> {
```

Comment:
I would probably keep this constraint as is and then just have:
```rust
store: Arc<S>
```

Reply:
I'm not sure I see the advantage.  Conceptually the provider cares only that it _has_ a `Store`, not the _kind_ of `Store` - what do we gain by requiring it to be generic in the type of store?

Reply:
Mostly I think of it as describing that we have a Provider of the given type with the store. A `WasccProvider<FileStore>` is a distinct provider and type from a `WasccProvider<InterceptingStore>` in my mind. However, using generics also avoids dynamic dispatching as it will have a concrete type

Reply:
![image](https://user-images.githubusercontent.com/865538/87089356-6db3d980-c28a-11ea-92d8-1890f355e448.png)